src/Microsoft.Performance.SDK.Runtime/ApplicationEnvironment.cs

```diff
@@ -148,5 +149,29 @@ private static MessageBoxIcon MapToImage(MessageType messageType)
                     return MessageBoxIcon.Information;
             }
         }
+
+        private sealed class NullTableSynchronizer
+            : ITableDataSynchronization
+        {
+            public void SubmitColumnChangeRequest(
+                IEnumerable<Guid> columns,
+                Action onReadyForChange,
+                Action onChangeComplete,
+                bool requestInitialFilterReevaluation = false)
+            {
+                onReadyForChange?.Invoke();
```

Comment:
according to the spec of this interface, these calls should happen in a lock

Reply:
Haha, I copied and pasted this from the runtime implementation.

Reply:
Actually, we shouldn't need the runtime implementation anymore given this change.

Reply:
Perhaps we should add the lock anyway :)

Reply:
Do you mean this comment?

`Lock tables affected by the change (based on <paramref name="predicate"/>) and wait for all existing operations on those tables to complete.`